src/react/components/shell/helpMenu.tsx

```diff
@@ -0,0 +1,131 @@
+                    displayName={strings.editorPage.help.title}
+                    accelerators={["Ctrl+H", "Ctrl+h"]}
+                    handler={() => this.setState({show: !this.state.show})}
+                    icon={this.icon}
+                    keyEventType={KeyEventType.KeyDown}
+                />
+                <MessageBox
+                    title={strings.titleBar.help}
+                    message={this.getHelpBody()}
+                    show={this.state.show}
+                    onCancel={this.onClose}
+                    hideFooter={true}
+                    closeOnOutsideClick={true}
+                />
+            </div>
+        );
+    }
+
+    private onClose = () => {
+        this.setState({show: false});
+        if (this.props.onClose) {
+            this.props.onClose();
+        }
+    }
+
+    private getHelpBody = () => {
+
+        const registrations = this.context.keyboard.getRegistrations()[KeyEventType.KeyDown];
+        if (!registrations) {
+            return;
+        }
+
+        const groupKeys = this.groupKeys(registrations);
+
+        return (
+            <div className="help-body container">
+                {
+                    groupKeys.map((group) => group.length ? this.getRegistrationRow(group, registrations) : null)
+                }
+            </div>
+        );
+    }
+
+    private groupKeys = (registrations: {[key: string]: IKeyboardBindingProps}) => {
+        const allKeys = Object.keys(registrations);
+        const caseConsolidatedKeys = this.consolidateKeyCasings(allKeys);
+
+        const groups = [];
+        const alreadyGrouped = new Set();
+
+        for (const key of caseConsolidatedKeys) {
+            const group = [key];
+            if (!alreadyGrouped.has(key)) {
+                alreadyGrouped.add(key);
+                for (const otherKey of caseConsolidatedKeys) {
+                    if (!alreadyGrouped.has(otherKey) &&
+                            this.bindingEquals(registrations[key], registrations[otherKey])) {
+                        group.push(otherKey);
+                        alreadyGrouped.add(otherKey);
+                    }
+                }
+                groups.push(group);
+            }
+        }
+        return groups;
+    }
+
+    private bindingEquals(binding1: IKeyboardBindingProps, binding2: IKeyboardBindingProps) {
+        return binding1 && binding2
+                && binding1.displayName === binding2.displayName
+                && binding1.handler === binding2.handler;
+    }
+
+    private consolidateKeyCasings = (allKeys: string[]): string[] => {
+        const lowerRegistrations = {};
+        for (const key of allKeys) {
+            const lowerKey = key.toLowerCase();
+            if (!lowerRegistrations[lowerKey]) {
+                lowerRegistrations[lowerKey] = key;
+            }
+        }
+        return Object.keys(lowerRegistrations).map((lowerKey) => lowerRegistrations[lowerKey]);
```

Comment:
I know discussed this offline and the reason to do this is to use the capitalize version.  But this is assuming that the capitalized version comes first.  Can we be sure that `const key of allKeys` return capitalize version first?

Reply:
We're not explicitly saying to only use the capital version, it's more just the first version listed, and I made all the keys capitalized. We just need to make sure the one we want to show is the first one we list in the keybindings